File: FragmentationX_core/src/main/java/me/liam/anim/FragmentAnimation.java
package me.liam.anim;

import me.liam.fragmentation.R;

public class FragmentAnimation {

    int enterAnimId;
    int exitAnimId;
    int popEnterAnimId;
    int popExitAnimId;

    public FragmentAnimation(int enterAnimId, int exitAnimId) {
        this(enterAnimId,exitAnimId,R.anim.anim_empty,R.anim.anim_empty);
    }

    public FragmentAnimation(int enterAnimId, int exitAnimId, int popEnterAnimId, int popExitAnimId) {
        this.enterAnimId = enterAnimId;
        this.exitAnimId = exitAnimId;
        this.popEnterAnimId = popEnterAnimId;
        this.popExitAnimId = popExitAnimId;
    }

    public int getEnterAnimId() {
        return enterAnimId;
    }

    public int getExitAnimId() {
        return exitAnimId;
    }

    public int getPopEnterAnimId() {
        return popEnterAnimId;
    }

    public int getPopExitAnimId() {
        return popExitAnimId;
    }

}

File: core/src/main/java/me/liam/fragmentation/anim/FragmentAnimation.kt
package me.liam.fragmentation.anim

import me.liam.fragmentation.R

open class FragmentAnimation @JvmOverloads constructor(enterAnimId: Int, exitAnimId: Int, popEnterAnimId: Int = R.anim.anim_empty, popExitAnimId: Int = R.anim.anim_empty) {

    var enterAnimId: Int = 0
        internal set
    var exitAnimId: Int = 0
        internal set
    var popEnterAnimId: Int = 0
        internal set
    var popExitAnimId: Int = 0
        internal set

    init {
        this.enterAnimId = enterAnimId
        this.exitAnimId = exitAnimId
        this.popEnterAnimId = popEnterAnimId
        this.popExitAnimId = popExitAnimId
    }

}
